ui.frontend/src/view/FormFieldBase.js

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+import Utils from "../utils";
+import FormField from './FormField'
+
+export default class FormFieldBase extends FormField {
+
+    constructor(params) {
+        super(params)
+        this.widget = this.getWidget();
+        this.description = this.getDescription();
+        this.label = this.getLabel();
+        this.errorDiv = this.getErrorDiv();
+        this.setId(this.element.id);
+    }
+
+    /**
+     * implementations should return the widget element that is used to capture the value from the user
+     * It will be a input/textarea element
+     * @returns
+     */
+    getWidget() {
+
+    }
+
+    /**
+     * implementations should return the element used to show the description of the field
+     * @returns
+     */
+    getDescription() {
+
+    }
+
+    /**
+     * implementations should return the element used to show the label of the field
+     * @returns
+     */
+    getLabel() {
+
+    }
+
+    /**
+     * implementations should return the element used to show the error on the field
+     * @returns
+     */
+    getErrorDiv() {
+
+    }
+
+    setModel(model) {
+        super.setModel(model);
+        const state = this._model.getState();
+        this._applyState(state);
+    }
+
+    /**
+     * applies full state of the field to the HTML. Generally done just after the model is bound to the field
+     * @param state
+     * @private
+     */
+    _applyState(state) {
```

Comment:
pls use # for private functions.` #applyState`

Reply:
https://github.com/tc39/proposal-private-fields/blob/main/FAQ.md#why-doesnt-thisx-access-the-private-field-named-x-given-that-thisx-does